lib/auth/webauthnwin/webauthn_windows.go

```diff
@@ -0,0 +1,603 @@
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	rp, err := rpToCType(in.Response.RelyingParty)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	u, err := userToCType(in.Response.User)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	credParam, err := credParamToCType(in.Response.Parameters)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	cd, jsonEncodedCD, err := clientDataToCType(in.Response.Challenge.String(), origin, string(protocol.CreateCeremony))
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	opts, err := n.makeCredOptionsToCType(in.Response)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	var out *webauthnCredentialAttestation
+	ret, _, lastErr := procWebAuthNAuthenticatorMakeCredential.Call(
+		uintptr(hwnd),
+		uintptr(unsafe.Pointer(rp)),
+		uintptr(unsafe.Pointer(u)),
+		uintptr(unsafe.Pointer(credParam)),
+		uintptr(unsafe.Pointer(cd)),
+		uintptr(unsafe.Pointer(opts)),
+		uintptr(unsafe.Pointer(&out)),
+	)
+	if ret != 0 {
+		return nil, trace.Wrap(getErrorNameOrLastErr(ret, lastErr))
+	}
+	if out == nil {
+		return nil, errors.New("unexpected nil response from MakeCredential")
+	}
+
+	// Note that we need to copy bytes out of `out` if we want to free object.
+	// That's why bytesFromCBytes is used.
+	// We don't care about free error so ignore it explicitly.
+	defer func() { _ = freeCredentialAttestation(out) }()
+
+	credential := bytesFromCBytes(out.cbCredentialId, out.pbCredentialId)
+
+	return &wanlib.CredentialCreationResponse{
+		PublicKeyCredential: wanlib.PublicKeyCredential{
+			Credential: wanlib.Credential{
+				ID:   base64.RawURLEncoding.EncodeToString(credential),
+				Type: string(protocol.PublicKeyCredentialType),
+			},
+			RawID: credential,
+		},
+		AttestationResponse: wanlib.AuthenticatorAttestationResponse{
+			AuthenticatorResponse: wanlib.AuthenticatorResponse{
+				ClientDataJSON: jsonEncodedCD,
+			},
+			AttestationObject: bytesFromCBytes(out.cbAttestationObject, out.pbAttestationObject),
+		},
+	}, nil
+}
+
+func freeCredentialAttestation(in *webauthnCredentialAttestation) error {
+	_, _, err := procWebAuthNFreeCredentialAttestation.Call(
+		uintptr(unsafe.Pointer(in)),
+	)
+	if err != syscall.Errno(0) {
+		return err
+	}
+	return nil
+}
+
+func freeAssertion(in *webauthnAssertion) error {
+	_, _, err := procWebAuthNFreeAssertion.Call(
+		uintptr(unsafe.Pointer(in)),
+	)
+	if err != syscall.Errno(0) {
+		return err
+	}
+	return nil
+}
+
+// checkIfDLLExistsAndGetAPIVersionNumber checks if dll exists and tries to load
+// it's version via API call. This function makes sure to not panic if dll is
+// missing.
+func checkIfDLLExistsAndGetAPIVersionNumber() (int, error) {
+	if err := modWebAuthn.Load(); err != nil {
+		return 0, err
+	}
+	if err := procWebAuthNGetApiVersionNumber.Find(); err != nil {
+		return 0, err
+	}
+	ret, _, err := procWebAuthNGetApiVersionNumber.Call()
+	if ret == 0 && err != syscall.Errno(0) {
+		return 0, err
+	}
```

Comment:
Could we do handle this like the others that do `if ret != 0 { return trace.Wrap(getErrorNameOrLastErr(...)) }` ?

Reply:
here API does not return `hresult` like in other API calls but returns dword, which is APi version. So in most cases it will be non 0 but this is fine.

More details: https://github.com/microsoft/webauthn/blob/7ab979cc833bfab9a682ed51761309db57f56c8c/webauthn.h#L895-L902

Reply:
Gotcha, maybe leave a comment? It's hard to spot the difference.

Reply:
added